packages/services/storage/src/index.ts

```diff
@@ -2511,6 +2548,37 @@ export async function createStorage(connection: string, maximumPoolSize: number)
           "id" = ${args.oidcIntegrationId}
       `);
     },
+
+    idMutex: {
+      async lock(id) {
+        const lockConn = await getLockConn();
+
+        const idInt = hashFnv32a(id);
+
+        // wait while there's a lock
+        let lock;
+        while ((lock = locks.get(id))) {
+          // first db then process
+          await lockConn.query(sql`select pg_advisory_lock(${idInt})`);
```

Comment:
What will this statement do, will it be blocked until the lock is released?
Will it raise an exception?

Reply:
I moved it below the while loop (and added more explanations), it must run only once per lock and should unlock the in-process lock if it fails - good catch!

https://github.com/kamilkisiela/graphql-hive/blob/b59aa864a9662dfab70375a43c2b0392cc27d488/packages/services/storage/src/index.ts#L2560-L2582